### PR TITLE
Add ApiOptions overloads to methods on I(Observable)StarredClient

### LINKED ISSUE #1163: Add ApiOption overloads to methods on INotificationsClient
As part of completing #760 a number of overloads need to be added to the client interfaces.

As an example, this has been added to [`IReleasesClient`](https://github.com/octokit/octokit.net/commit/03ea83148995711fc47148489e2123e65f33bd82)and [`IObservableReleasesClient`](https://github.com/octokit/octokit.net/commit/aa3ce0db9210b778414e290af7f17c4a646bdeec).

The checklist for porting each method is:

- Add an overload for this method on `INotificationsClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `INotificationsClient`
- Add an overload for this method on `IObservableNotificationsClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IObservableNotificationsClient`

This needs to be done for each of these methods:

- [ ] `Task<IReadOnlyList<Notification>> GetAllForCurrent()`
- [ ] `Task<IReadOnlyList<Notification>> GetAllForCurrent(NotificationsRequest request)`
- [ ] `Task<IReadOnlyList<Notification>> GetAllForRepository(String owner, String name)`
- [ ] `Task<IReadOnlyList<Notification>> GetAllForRepository(String owner, String name, NotificationsRequest request)`

### PR DESCRIPTION
**Pull request by dampir**
Fixes #1163 

To-do:

- [x]  Add overloads on IStarredClient
- [x]  Add overloads on IObservableStarredClient
- [x]  Add unit tests for these new methods
- [x]  Add integration tests for these new methods

/cc @shiftkey, @ryangribble

### CODE REVIEW

File: Octokit.Reactive/Helpers/ConnectionExtensions.cs [hunk 1]
```diff
@@ -37,6 +37,15 @@ public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connectio
             return GetPages(url, parameters, (pageUrl, pageParams) => connection.Get<List<T>>(pageUrl, pageParams, accepts).ToObservable());
         }
 
+        public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connection, Uri url, IDictionary<string, string> parameters, string accepts, ApiOptions options)
+        {
+            return GetPagesWithOptions(url, parameters, options, (pageUrl, pageParams, o) =>
```
Comment (shiftkey) [inline Octokit.Reactive/Helpers/ConnectionExtensions.cs]: This method isn't paginating as I'd hoped - it's continuing to make the same request forever. Looking into what I've missed here. Here's the test I'm using, feel free to pull it into this PR:

```
[IntegrationTest]
public async Task CanRetrievePagesOfPublicStarWatchers()
{
    var owner = "octokit";
    var name = "octokit.net";

    var firstPageOptions = new ApiOptions
    {
        PageSize = 5,
        StartPage = 1
    };

    var firstPage = await _fixture.GetAllStargazersWithTimestamps(owner, name, firstPageOptions);

    var secondPageOptions = new ApiOptions
    {
        PageSize = 5,
        StartPage = 2
    };

    var secondPage = await _fixture.GetAllStargazersWithTimestamps(owner, name, secondPageOptions);
    Assert.Equal(5, firstPage.Count);
    Assert.Equal(5, secondPage.Count);

    Assert.NotEqual(firstPage[0].User.Id, secondPage[0].User.Id);
    Assert.NotEqual(firstPage[1].User.Id, secondPage[1].User.Id);
    Assert.NotEqual(firstPage[2].User.Id, secondPage[2].User.Id);
    Assert.NotEqual(firstPage[3].User.Id, secondPage[3].User.Id);
    Assert.NotEqual(firstPage[4].User.Id, secondPage[4].User.Id);
}
```

**Comment by shiftkey** [comment]
@dampir this looks good, add in that new integration test and this is good to :ship:

File: Octokit.Reactive/Helpers/ConnectionExtensions.cs (hunk 1 above)
Comment (dampir) [inline Octokit.Reactive/Helpers/ConnectionExtensions.cs]: @shiftkey, sorry, I've forgot push my commits with integration tests. There are a lot of new ones :+1:

Comment (shiftkey) [inline Octokit.Reactive/Helpers/ConnectionExtensions.cs]: @dampir good to hear :heart:

**Comment by dampir** [comment]
@shiftkey please rerun Travis, it failed again :cry:

**Pull request merged**